### ISSUE TITLE
Director service skips one spec file

### ISSUE THREAD

**Issue by romankhomitskyi** [sorry-cypress/sorry-cypress#543]
Please note - issues that are not following the template will be closed automatically:

- [x] I have verified that all sorry-cypress services are running and accessible
- [x] I have read documentation and found no answer to my problem
- [x] I have searched issues / discussions and found to answer to my problem
- [x] I have read [Minio Configutation guide](https://docs.sorry-cypress.dev/configuration/director-configuration/minio-configuration) - for issues related to Screenshots / Videos uploads for Minio 
- [ ] I have activated the debug mode `DEBUG=cypress:server:api cy2 ....` and have the network error

## Summary
First I change url using patch function then  I run my tests using the following command
```
 yarn cypress run --browser chrome --parallel --record --key somekey1 --ci-build-id n8
```
Even though I passing different `--ci-build-id`, I see that run does not include one spec(see screenshots), I guess it randomly skips one spec file

My set up only includes director service

```
cypress-director:
    image: agoldis/sorry-cypress-director:latest
    container_name: "cypress-director"
    environment:
      DASHBOARD_URL: http://localhost:8080
      SCREENSHOTS_DRIVER: ../screenshots/dummy.driver
    networks:
      - home
```
I have tried to restart & rebuild director service, I have killed all containers

## How to reproduce

1. Use only director service and  run `yarn cypress run --browser chrome --parallel --record --key somekey1 --ci-build-id n8`

## Relevant logs / screenshots

![image](https://user-images.githubusercontent.com/57874507/155998827-2943b9df-ddc9-468a-ac53-ceac11b5ac9e.png)

## Environment

- sorry-cypress version: `<2.1.1>`
- cypress version: `<9.5.0>`
- platform: `docker`
- service: `director`

**Comment by agoldis** [comment]
@romankhomitskyi does this issue occur all the time or intermittent? I just tried and could not reproduce it. 
How many parallel cypress runners are you using?

**Comment by romankhomitskyi** [comment]
> @romankhomitskyi does this issue occur all the time or intermittent? I just tried and could not reproduce it. How many parallel cypress runners are you using?

This issue occurs all the time
I use 9 parallel cypress runners in one container
Hovewer it occurs even when I do not run tests in parallel

**Comment by romankhomitskyi** [comment]
Long time ago I used full setup, then I decided to use only director service
I deleted all mongo db data from project and do not use any others services any more

My CI for each pipeline creates a brand new services (clear environment) unfortunately issue is still there

**Comment by romankhomitskyi** [comment]
<img width="132" alt="Screenshot 2022-03-03 at 09 18 49" src="https://user-images.githubusercontent.com/57874507/156515649-34969655-3020-48d2-8839-4cd55e02d0d5.png">

**Comment by romankhomitskyi** [comment]
Ok, it skips the first spec all the time 
I do not have this issue when running tests without `--parallel --record --key somekey1`

**Comment by romankhomitskyi** [comment]
Hmm now I don't think it skips spec, I see the first spec in report
I can confirm that it's just visual confusing starting run from 2 spec
It has no affect on spec run itself

**Comment by agoldis** [comment]
@romankhomitskyi "starting run from 2 spec" do you mean from the 2nd spec?

**Comment by romankhomitskyi** [comment]
> @romankhomitskyi "starting run from 2 spec" do you mean from the 2nd spec?

I meant,  it shows running 2 of 90,  however it should be 1 of 90

**Comment by Fryuni** [comment]
This seems related to the problem we are experiencing. We created a workflow that start 25 machines to run our tests in parallel. The tests are being executed and reported properly, but most of the time one of the specs, seemingly at random, doesn't even start.

We've checked the logs on all the VMs multiple times and the spec is simply not assigned to any of them
![image](https://user-images.githubusercontent.com/11063910/159135439-19d6fea4-6aad-4f92-9c59-1baef6d9b4fe.png)

For us this is an intermittent behavior, happening about 60% of the time (we ran the same workflow about 200 times to check). It sometimes happens only for one of the groups (we have chrome and firefox) and sometimes one spec for each.

![image](https://user-images.githubusercontent.com/11063910/159135514-64796b8a-ebcd-4d82-ba50-fb02a49a3c3b.png)

**Comment by agoldis** [comment]
@Fryuni it is a different issue - you're using a different driver and groups. Is seems like there's some kind of race condition that breaks parallelization when using groups.

**Comment by Fryuni** [comment]
I'll collect the logs from SC and cy2 as soon as I have time and open a new issue then

**Comment by marcospassos** [comment]
@agoldis, this issue is blocking us from using Sorry Cypress – every run fails. It became more critical once we grew our number of tests. Do you have any clue where we should start looking for the issue? The problem is probably related to the director, but any guess which path can be causing the issue?

**Comment by marcospassos** [comment]
It's also broken on currents.dev:
<img width="1113" alt="image" src="https://user-images.githubusercontent.com/943036/159326891-4d048101-2c75-4153-b66b-363ca301f782.png">

We're opening a PR with more details.

**Comment by agoldis** [comment]
@marcospassos Currents implementation is a bit different from sorry-cypress, it is optimized for higher scale and uses redis, while sorry-cypress uses mongo. 

There's some material available here for timeouts: https://currents.dev/readme/runs/run-timeouts

--- 

@marcospassos are you also experiencing issues when parallelizing grouped tests? 


In general, it'd be great to see the logs from director service and also the state of the database for specific run - if one of the spec files is never "claimed" it's likely to be some nasty race condition.

There's a also a chance it's something related to your specific setup, considering 150k+ [weekly runs](https://www.npmjs.com/package/cy2), I'd expect more complaints.

**Comment by elena-altuhova** [comment]
Experiencing the same issue with basic setup without mongo DB. Correct amount of features is identified and then run starts  considering first feature as 2nd. 
![image](https://user-images.githubusercontent.com/33543339/180189332-871c06f3-345e-40ce-b510-0a6710fff59c.png)
Running:
